<a href="https://colab.research.google.com/github/CameronBannick/Sales_Predictions/blob/main/project_2_part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Proposal 1. UFO SIGHTINGS from Kaggle

In [ ]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
ufo_df = pd.read_csv('/content/drive/MyDrive/Coding Dojo/Week 7/Data Proposals/nuforc_reports.csv')
ufo_df.head()

summary        city state  \
0  My wife was driving southeast on a fairly popu...     Chester    VA   
1  I think that I may caught a UFO on the NBC Nig...  Rocky Hill    CT   
2  I woke up late in the afternoon 3:30-4pm. I we...         NaN   NaN   
3  I was driving towards the intersection of fall...      Ottawa    ON   
4  In Peoria Arizona, I saw a cigar shaped craft ...      Peoria    NY   

             date_time     shape     duration  \
0  2019-12-12T18:43:00     light    5 seconds   
1  2019-03-22T18:30:00    circle  3-5 seconds   
2                  NaN       NaN          NaN   
3  2019-04-17T02:00:00  teardrop   10 seconds   
4  2009-03-15T18:00:00     cigar    2 minutes   

                                               stats  \
0  Occurred : 12/12/2019 18:43  (Entered as : 12/...   
1  Occurred : 3/22/2019 18:30  (Entered as : 03/2...   
2  Occurred : 4/1/2019 15:45  (Entered as : April...   
3  Occurred : 4/17/2019 02:00  (Entered as : 04-1...   
4  Occurred : 3/15/2009 18:00  (Entered as : 03/1...   

                                         report_link  \
0  http://www.nuforc.org/webreports/151/S151739.html   
1  http://www.nuforc.org/webreports/145/S145297.html   
2  http://www.nuforc.org/webreports/145/S145556.html   
3  http://www.nuforc.org/webreports/145/S145697.html   
4  http://www.nuforc.org/webreports/145/S145723.html   

                                                text               posted  \
0  My wife was driving southeast on a fairly popu...  2019-12-22T00:00:00   
1  I think that I may caught a UFO on the NBC Nig...  2019-03-29T00:00:00   
2  I woke up late in the afternoon 3:30-4pm.  I w...                  NaN   
3  I was driving towards the intersection of fall...  2019-04-18T00:00:00   
4  In Peoria, Arizona, I saw a cigar shaped craft...  2019-04-18T00:00:00   

   city_latitude  city_longitude  
0      37.343152      -77.408582  
1      41.664800      -72.639300  
2            NaN             NaN  
3      45.381383      -75.708501  
4            NaN             NaN

In [ ]:
ufo_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88125 entries, 0 to 88124
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   summary         88095 non-null  object 
 1   city            87891 non-null  object 
 2   state           82890 non-null  object 
 3   date_time       86938 non-null  object 
 4   shape           85627 non-null  object 
 5   duration        84954 non-null  object 
 6   stats           88088 non-null  object 
 7   report_link     88125 non-null  object 
 8   text            88070 non-null  object 
 9   posted          86938 non-null  object 
 10  city_latitude   72013 non-null  float64
 11  city_longitude  72013 non-null  float64
dtypes: float64(2), object(10)
memory usage: 8.1+ MB


First choice: dataset 1

1. Source of data: Kaggle

2. Brief description of data: A collection of UFO sightings based on UFO shaoe and location

3. What is the target? State

4. Is this a classification or regression problem? Classification

5. How many features? 12

6. How many rows of data. 88125

7. What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset? Missing data, i completly understand this might not be the best example for a beginner but it was too cool for me not to propose

### Proposal 2: Mall Report

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Coding Dojo/Week 7/Data Proposals/Mall_Customers.csv')
df.head()

CustomerID   Genre  Age  Annual Income (k$)  Spending Score (1-100)
0           1    Male   19                  15                      39
1           2    Male   21                  15                      81
2           3  Female   20                  16                       6
3           4  Female   23                  16                      77
4           5  Female   31                  17                      40

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   CustomerID              200 non-null    int64 
 1   Genre                   200 non-null    object
 2   Age                     200 non-null    int64 
 3   Annual Income (k$)      200 non-null    int64 
 4   Spending Score (1-100)  200 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 7.9+ KB


1. Source of data: Kaggle

2. Brief description of data: List of Mall Customers with age and incomes

3. What is the target? Spending Score

4. Is this a classification or regression problem? Regression

5. How many features? 5

6. How many rows of data. 200

7. What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset? It might be too simple to create a properly fitted model.